src/main/java/org/jabref/logic/importer/ImportFormatReader.java

```diff
@@ -215,4 +218,39 @@ public UnknownFormatImport importUnknownFormat(Path filePath) throws ImportExcep
 
         throw new ImportException(Localization.lang("Could not find a suitable import format."));
     }
+
+    public UnknownFormatImport importUnknownFormatFromString(String data) throws ImportException {
+        // stores ref to best result, gets updated at the next loop
+        List<BibEntry> bestResult = null;
+        long bestResultCount = 0;
+        String bestFormatName = null;
+
+        // Cycle through all importers:
+        for (Importer imFo : getImportFormats()) {
+            try (StringReader in = new StringReader(data); BufferedReader input = new BufferedReader(in)) {
+                ParserResult parserResult = imFo.importDatabase(input);
```

Comment:
Can you please extract the method `Importer.importDatabase(String data)`. This new method can be reused at least in the `MrDLibFetcher`. Similarly, for the `isRecognizedFormat`.